qa-tests-backend/src/main/groovy/services/BaseService.groovy

```diff
@@ -104,41 +104,43 @@ class BaseService {
         }
     }
 
-    static ManagedChannel transportChannel = null
-    static ClientInterceptor authInterceptor = null
-    static Channel effectiveChannel = null
-    private static boolean useClientCert = false
+    // codenarc-disable FieldName
+    private static final ThreadLocal<ManagedChannel> transportChannel = ThreadLocal.withInitial(() -> null)
+    private static final ThreadLocal<ClientInterceptor> authInterceptor = ThreadLocal.withInitial(() -> null)
+    private static final ThreadLocal<Channel> effectiveChannel = ThreadLocal.withInitial(() -> null)
+    private static final ThreadLocal<boolean> useClientCert = ThreadLocal.withInitial(() -> false)
+    // codenarc-enable FieldName
 
     static initializeChannel() {
-        if (transportChannel == null) {
+        if (transportChannel.get() == null) {
             SslContextBuilder sslContextBuilder = GrpcSslContexts
                     .forClient()
                     .trustManager(InsecureTrustManagerFactory.INSTANCE)
-            if (useClientCert) {
+            if (useClientCert.get()) {
                 sslContextBuilder = sslContextBuilder.keyManager(Keys.keyManagerFactory())
             }
             def sslContext = sslContextBuilder.build()
 
-            transportChannel = NettyChannelBuilder
+            transportChannel.set(NettyChannelBuilder
                     .forAddress(Env.mustGetHostname(), Env.mustGetPort())
                     .enableRetry()
                     .negotiationType(NegotiationType.TLS)
                     .sslContext(sslContext)
-                    .build()
-            effectiveChannel = null
+                    .build())
+            effectiveChannel.set(null)
         }
 
-        if (authInterceptor == null) {
-            effectiveChannel = transportChannel
+        if (authInterceptor.get() == null) {
+            effectiveChannel.set(transportChannel.get())
         } else {
-            effectiveChannel = ClientInterceptors.intercept(transportChannel, authInterceptor)
+            effectiveChannel.set(ClientInterceptors.intercept(transportChannel.get(), authInterceptor.get()))
         }
     }
 
     static Channel getChannel() {
-        if (effectiveChannel == null) {
+        if (effectiveChannel.get() == null) {
```

Comment:
Why do we do it lazy. We could call `initializeChannel` in static initializer for this class although I think it will be better to remove all statics and make it normal class that will be created in BaseSpec and used as a filed.

Reply:
@janisz can you re-review? I don't really want to stuck on more BaseService refactoring at this point. See the 55 calls to BaseService.* for instance.